src/wms/src/test/java/org/geoserver/wms/wms_1_1_1/CapabilitiesTest.java

```diff
@@ -795,6 +796,223 @@ public void testNonAdvertisedNonQueriableWithinGroup() throws Exception {
+
+        try {
+            layer.getMetadata().put(PublishedInfo.ROOT_IN_CAPABILITIES, true);
+            catalog.save(layer);
+            Document dom = dom(get("cite/Lakes/wms?request=GetCapabilities&version=1.1.0"), true);
+            // print(dom);
+
+            assertXpathEvaluatesTo("Lakes", "/WMT_MS_Capabilities/Capability/Layer/Name", dom);
+        } finally {
+            layer.getMetadata().put(PublishedInfo.ROOT_IN_CAPABILITIES, null);
+            catalog.save(layer);
+            setAdvertised(catalog, MockData.LAKES, true);
+            setQueryable(catalog, MockData.LAKES, true);
+        }
+    }
+
+    @Test
+    public void testRootLayerRemovedRequestParam() throws Exception {
+        // make layers non advertised
+        Catalog catalog = getCatalog();
+        setAdvertised(catalog, MockData.LAKES, false);
+        setQueryable(catalog, MockData.LAKES, false);
+        setAdvertised(catalog, MockData.NAMED_PLACES, false);
+        setQueryable(catalog, MockData.NAMED_PLACES, false);
+        LayerGroupInfo group = null;
+        try {
+            group = createLakesPlacesLayerGroup(catalog, LayerGroupInfo.Mode.NAMED, null);
+            Document dom =
+                    dom(
+                            get(
+                                    "lakes_and_places/wms?request=GetCapabilities&version=1.1.0&noRootLayer=true"),
+                            true);
+            // print(dom);
+
+            assertXpathEvaluatesTo(
+                    "lakes_and_places", "/WMT_MS_Capabilities/Capability/Layer/Name", dom);
+        } finally {
+            if (group != null) {
+                catalog.remove(group);
+            }
+            setAdvertised(catalog, MockData.LAKES, true);
+            setQueryable(catalog, MockData.LAKES, true);
+            setAdvertised(catalog, MockData.NAMED_PLACES, true);
+            setQueryable(catalog, MockData.NAMED_PLACES, true);
+        }
+    }
+
+    @Test
+    public void testRootLayerRemovedRequestParamHasGreaterPriority() throws Exception {
+        // make layers non advertised
+        WMSInfo info = getWMS().getServiceInfo();
+        info.getMetadata().put(WMS.NO_ROOT_LAYER_IN_CAPABILITIES_KEY, true);
+        Catalog catalog = getCatalog();
+        setAdvertised(catalog, MockData.LAKES, false);
+        setQueryable(catalog, MockData.LAKES, false);
+        setAdvertised(catalog, MockData.NAMED_PLACES, false);
+        setQueryable(catalog, MockData.NAMED_PLACES, false);
+        LayerGroupInfo group = null;
+        try {
+            group = createLakesPlacesLayerGroup(catalog, LayerGroupInfo.Mode.NAMED, null);
+            group.getMetadata().put(PublishedInfo.ROOT_IN_CAPABILITIES, true);
+            Document dom =
+                    dom(
+                            get(
+                                    "lakes_and_places/wms?request=GetCapabilities&version=1.1.0&noRootLayer=false"),
```

Comment:
``noRootLayer=false`` is hard to read imho, it means that the root layer is there, double negation.
``rootLayer=true``, ``rootLayer=false`` seem to read better to me.